src/btf.cpp

```diff
@@ -707,11 +693,91 @@ std::map<std::string, std::vector<std::string>> BTF::get_params_from_btf(
   return params;
 }
 
-std::map<std::string, std::vector<std::string>> BTF::get_params(
-    const std::set<std::string> &funcs,
-    bool is_raw_tracepoint) const
+FuncParamLists BTF::get_raw_tracepoints_params_from_btf(
+    const BTFObj &btf_obj,
+    const std::set<std::string> &rawtracepoints) const
+{
+  __s32 id, max = static_cast<__s32>(type_cnt(btf_obj.btf));
+  std::string type = std::string("");
+  struct btf_dump *dump;
+  char err_buf[256];
+  int err;
+
+  dump = dump_new(btf_obj.btf, dump_printf, &type);
+  err = libbpf_get_error(dump);
+  if (err) {
+    libbpf_strerror(err, err_buf, sizeof(err_buf));
+    LOG(ERROR) << "BTF: failed to initialize dump (" << err_buf << ")";
+    return {};
+  }
+
+  FuncParamLists params;
+  for (id = start_id(btf_obj.btf); id <= max; id++) {
```

Comment:
The entire BTF iteration is cumbersome, it would be great if we could simplify it one day. Definitely out of the scope of this PR, though.

Reply:
Yeah, it's crazy how many odd bugs and things I ran into trying to refactor BTF loading